src/coreclr/src/nativeaot/System.Private.CoreLib/src/Internal/DeveloperExperience/DeveloperExperience.cs

```diff
@@ -36,10 +33,17 @@ private static bool IsMetadataStackTraceResolutionDisabled()
             return disableMetadata;
         }
 
-        public virtual void WriteLine(string s)
+        internal static void WriteStackTraceLine(string s)
         {
-            Debug.WriteLine(s);
-            return;
+            Action<string> writeLineMethod = AppContext.GetData("System.Runtime.ExceptionServices.WriteStackTraceString") as Action<string>;
+            if (writeLineMethod != null)
+            {
+                writeLineMethod(s);
+            }
+            else
+            {
+                Debug.WriteLine(s);
+            }
         }
 
         public virtual string CreateStackTraceString(IntPtr ip, bool includeFileInfo)
```

Comment:
This does not need to be public with virtual methods anymore.

Reply:
It does if we want to bring back dotnet/corert#7658 at some point.